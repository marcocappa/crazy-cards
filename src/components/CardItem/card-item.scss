.card-detail {
    &__name {
        align-items: center;
        display: flex;
        margin-top: 0;

        & svg {
            margin-right: 0.5rem;
            width: 1.25rem;
        }
    }

    &__info {
        font-weight: 300;
        span {
            font-weight: 600;
        }
    }
}
