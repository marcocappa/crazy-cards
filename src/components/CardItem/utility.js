import React from 'react';
import { IconAnywhere, IconLiquid, IconStudent } from '../Icons';

export const getCardIcon = (id) => {
  switch (id) {
    case 'card-0':
      return <IconStudent />;
    case 'card-1':
      return <IconAnywhere />;
    case 'card-2':
      return <IconLiquid />;
    default:
      return <IconLiquid />;
  }
};
