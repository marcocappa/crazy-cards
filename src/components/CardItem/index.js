import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import { getCardIcon } from './utility';
import './card-item.scss';

const CardItem = ({ className, id, name, apr, btod, pod, creditAvailable }) => (
  <li className={`card-detail ${classnames(className)}`}>
    <h3 className="card-detail__name">
      {getCardIcon(id)}
      {name}
    </h3>
    <p className="card-detail__info card-detail__info--apr">
      Representative % APR (variable): <span>{apr}%</span>
    </p>
    <p className="card-detail__info card-detail__info--balance">
      Balance Transfer offer duration: <span>{btod} months</span>
    </p>
    <p className="card-detail__info card-detail__info--purchase">
      Purchase offer duration: <span>{pod} months</span>
    </p>
    <p className="card-detail__info card-detail__info--credit">
      Credit Available: <span>Â£{creditAvailable}</span>
    </p>
  </li>
);

CardItem.propTypes = {
  className: PropTypes.string,
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  apr: PropTypes.number.isRequired,
  btod: PropTypes.number.isRequired,
  pod: PropTypes.number.isRequired,
  creditAvailable: PropTypes.number.isRequired,
};

CardItem.defaultProps = {
  className: '',
};

export default CardItem;
