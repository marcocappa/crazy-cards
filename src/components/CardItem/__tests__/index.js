import React from 'react';
import CardItem from '../index';
import { shallow } from 'enzyme';

describe('<CardItem />', () => {
  it('should match snapshot', () => {
    const mockProps = {
      className: 'additional-class',
      id: 'card-0',
      name: 'Student Life',
      apr: 18.9,
      btod: 0,
      pod: 6,
      creditAvailable: 1200,
    };

    const wrapper = shallow(<CardItem {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
