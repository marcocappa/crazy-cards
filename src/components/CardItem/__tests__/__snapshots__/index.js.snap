// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardItem /> should match snapshot 1`] = `
<li
  className="card-detail additional-class"
>
  <h3
    className="card-detail__name"
  >
    <IconStudent />
    Student Life
  </h3>
  <p
    className="card-detail__info card-detail__info--apr"
  >
    Representative % APR (variable): 
    <span>
      18.9
      %
    </span>
  </p>
  <p
    className="card-detail__info card-detail__info--balance"
  >
    Balance Transfer offer duration: 
    <span>
      0
       months
    </span>
  </p>
  <p
    className="card-detail__info card-detail__info--purchase"
  >
    Purchase offer duration: 
    <span>
      6
       months
    </span>
  </p>
  <p
    className="card-detail__info card-detail__info--credit"
  >
    Credit Available: 
    <span>
      Â£
      1200
    </span>
  </p>
</li>
`;
